.travis.yml

```diff
@@ -37,8 +37,6 @@ addons:
 
 # Install various dependencies
 install:
-  - pip install --upgrade pytest
-  - pip install --upgrade pytest-cov
```

Comment:
Even if you keep those lines, Spack won't use the pip versions.  Externals come before the system paths.  We could change this to something similar to what I briefly did with cyaml, but if you remember, using the different yaml version on some systems caused unexpected bugs.  This probably is not as important for `spack test`, but think about what happens if the latest version of `pytest` drops Python 2.6 support.  Our tests will start breaking before we can update.  I'd rather keep the externals for that reason, and one other: while I do think developers should be able to install pytest, I *really* want to lower the barriers to contributing, so I don't think they should have to.